src/runtime-rs/crates/hypervisor/src/device/topology.rs

```diff
@@ -225,15 +304,35 @@ impl PCIeTopology {
             root_bus_devices: HashMap::with_capacity(PCIE_ROOT_BUS_SLOTS_CAPACITY as usize),
         };
 
+        // initialize port devices within PCIe Topology
+        let total_rp: usize = topo_config.device_info.pcie_root_port as usize;
+        let total_sp: usize = topo_config.device_info.pcie_switch_port as usize;
+
+        let mut devices_per_port: HashMap<PCIePort, Vec<TopologyPortDevice>> = HashMap::new();
```

Comment:
Is it really necessary to always insert both `PCIePort` types even if we have the following logic below:

```rust
(r, s) if r > 0 && s > 0 => PCIePort::InvalidPort,
```

I am not sure, but it would be better to place a validity check before so that we could prepare either of them.

Reply:
validity check logic added